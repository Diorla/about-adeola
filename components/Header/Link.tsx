import React from "react";
import Link from "next/link";
import styled from "styled-components";

export const StyledLink = styled.a<{ active?: boolean }>`
  color: ${({ active }) => (active ? "#ffffff" : "#ffffffcc")};
  font-size: 48px;
  transition: 0.3s linear;
  transform-origin: top;
  &:hover {
    text-decoration: none;
    color: #ffffff;
    transform: skewY(5deg);
  }
`;

const HeaderLink = ({
  href,
  title,
  active,
}: {
  href: string;
  title: string;
  active?: boolean;
}) => (
  <Link href={href}>
    <StyledLink active={active}>{title}</StyledLink>
  </Link>
);

export default HeaderLink;
