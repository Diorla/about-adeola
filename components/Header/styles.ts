import { Header } from "styled-components";
import breakpoints from "../../theme/breakpoints";
import elevation from "../../theme/elevation";
import palette from "../../theme/palette";
import priority from "../../theme/priority";

const styles: Header = {
  link: {
    color: palette.primary.main,
    cursor: "url(/cursor/pointer.svg), pointer",
    padding: "4px",
    margin: "4px",
  },
  logo: {
    fontFamily: '"Oleo Script", cursive',
    fontSize: "24px",
    textDecoration: "none",
    margin: "0",
    padding: "0",
  },
  contact: {
    backgroundColor: palette.primary.main,
    borderRadius: "4px",
    textAlign: "center",
    hoverColor: palette.primary.light,
    color: palette.primary.text,
    textDecoration: "none",
  },
  main: {
    display: "flex",
    justifyContent: "space-between",
    padding: "8px",
    alignItems: "center",
    position: "fixed",
    top: 0,
    zIndex: priority.header,
    backgroundColor: "white",
    boxShadow: elevation[1],
    breakpoint: breakpoints.md,
    mdDirection: "column",
    mdJustifyContent: "space-evenly",
    mdPosition: "absolute",
    width: "100%",
  },
};

export default styles;
