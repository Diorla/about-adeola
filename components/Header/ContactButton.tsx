import React from "react";
import Link from "next/link";
import styled from "styled-components";
import timing from "../../theme/timing";
import { StyledLink } from "./Link";
import styles from "./styles";

const { contact } = styles;

export const StyledHighlight = styled(StyledLink)`
  background: ${contact.backgroundColor};
  border-radius: ${contact.borderRadius};
  text-align: ${contact.textAlign};
  color: ${contact.color};
  transition: ${timing.transition};
  &:hover {
    background: ${contact.hoverColor};
    text-decoration: ${contact.textDecoration};
  }
`;

export default () => (
  <Link href="#contact">
    <StyledHighlight>Contact</StyledHighlight>
  </Link>
);
