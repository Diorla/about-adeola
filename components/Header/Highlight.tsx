import Link from "next/link";
import styled from "styled-components";
import animation from "../../theme/animation";
import { StyledLink } from "./Link";
import styles from "./styles";

const { highlight } = styles;

export const StyledHighlight = styled(StyledLink)`
  background: ${highlight.backgroundColor};
  border-radius: ${highlight.borderRadius};
  text-align: ${highlight.textAlign};
  color: ${highlight.color};
  transition: ${animation.transition};
  &:hover {
    background: ${highlight.hoverColor};
    text-decoration: ${highlight.textDecoration};
  }
`;

export default ({ href, title }: { href: string; title: string }) => (
  <Link href={href}>
    <StyledHighlight>{title}</StyledHighlight>
  </Link>
);
