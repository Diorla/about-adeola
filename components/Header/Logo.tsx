import Link from "next/link";
import styled from "styled-components";
import { StyledLink } from "./Link";
import styles from "./styles";

const { logo } = styles;

export const StyledLogo = styled(StyledLink)`
  font-family: ${logo.fontFamily};
  font-size: 28px;
  color: white;
  font-weight: bold;
  text-shadow: 0 0 2px black;
  & > span {
    opacity: 0;
    transition: 0.4s linear;
  }
  & > span:first-child {
    transition: 0.6s linear;
  }
  & > span:last-child {
    transition: 0.2s linear;
  }
  &:hover {
    transform: none;
    font-weight: bolder;
  }
  &:hover > span {
    opacity: 1;
  }
`;

const Logo = () => (
  <Link href="/">
    <StyledLogo>
      <span>{"<"}</span>Code<span>{" /"}</span>
      <span>{">"}</span>
    </StyledLogo>
  </Link>
);

export default Logo;
