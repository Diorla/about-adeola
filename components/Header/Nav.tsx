import React from "react";
import styled from "styled-components";
import styles from "./styles";

const { main } = styles;
export const StyledNav = styled.nav`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  justify-content: space-evenly;
  @media screen and (max-width: ${main.breakpoint}) {
    width: 100%;
  }
`;

export default ({ children }: { children: React.ReactNode }) => (
  <StyledNav>{children}</StyledNav>
);
