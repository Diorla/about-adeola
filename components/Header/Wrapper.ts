import styled from "styled-components";
import styles from "./styles";

const { main } = styles;

export default styled.div`
  display: ${main.display};
  justify-content: ${main.justifyContent};
  padding: ${main.padding};
  align-items: ${main.alignItems};
  position: ${main.position};
  top: ${main.top};
  z-index: ${main.zIndex};
  background: ${main.backgroundColor};
  box-shadow: ${main.boxShadow};
  width: ${main.width};
   @media screen and (max-width: ${main.breakpoint}) {
    flex-direction: ${main.mdDirection};
    justify-content: ${main.mdJustifyContent};
    position: ${main.mdPosition};
  }
`;
