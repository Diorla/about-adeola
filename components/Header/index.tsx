import React from "react";
import Link from "./Link";
import Logo from "./Logo";
import ContactButton from "./ContactButton";
import Nav from "./Nav";
import Wrapper from "./Wrapper";

export default () => (
  <Wrapper id="top">
    <Logo />
    <Nav>
      <Link href="/about" title="About" />
      <Link href="/showcase" title="Showcase" />
      <Link href="/resources" title="Resources" />
      <ContactButton />
    </Nav>
  </Wrapper>
);
