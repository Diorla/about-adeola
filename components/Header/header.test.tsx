import React from "react";
import renderer from "react-test-renderer";
import "ts-jest";
import "jest-styled-components";
import Header from "./index";
import ContactButton from "./ContactButton";
import Logo from "./Logo";
import Link from "./Link";
import Nav from "./Nav";
import Wrapper from "./Wrapper";

describe("Testing components", () => {
  test("Contact me button", () => {
    const tree = renderer.create(<ContactButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Header", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders links appropriately", () => {
    const tree = renderer.create(<Link href="/about" title="About" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Logo", () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Nav", () => {
    const tree = renderer
      .create(
        <Nav>
          <Link href="/about" title="About" />
        </Nav>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Header wrapper", () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
