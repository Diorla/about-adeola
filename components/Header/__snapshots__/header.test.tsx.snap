// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components Contact me button 1`] = `
.c0 {
  color: #008080;
  cursor: pointer;
  margin: 4px;
  padding: 4px;
}

.c1 {
  background: #008080;
  border-radius: 4px;
  text-align: center;
  color: #fff;
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
}

.c1:hover {
  background: #4cb0af;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0 c1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Contact
</a>
`;

exports[`Testing components Header 1`] = `
.c1 {
  color: #008080;
  cursor: pointer;
  margin: 4px;
  padding: 4px;
}

.c2 {
  font-family: "Oleo Script",cursive;
  font-size: 24px;
  margin: 0;
  padding: 0;
}

.c2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  background: #008080;
  border-radius: 4px;
  text-align: center;
  color: #fff;
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
}

.c4:hover {
  background: #4cb0af;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0;
  z-index: 1000;
  background: white;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  width: 100%;
}

@media screen and (max-width:768px) {
  .c3 {
    width: 100%;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    position: absolute;
  }
}

<div
  className="c0"
  id="top"
>
  <a
    className="c1 c2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Adeola Ade
  </a>
  <nav
    className="c3"
  >
    <a
      className="c1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      About
    </a>
    <a
      className="c1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Showcase
    </a>
    <a
      className="c1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Resources
    </a>
    <a
      className="c1 c4"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Contact
    </a>
  </nav>
</div>
`;

exports[`Testing components Header wrapper 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0;
  z-index: 1000;
  background: white;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  width: 100%;
}

@media screen and (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    position: absolute;
  }
}

<div
  className="c0"
/>
`;

exports[`Testing components Logo 1`] = `
.c0 {
  color: #008080;
  cursor: pointer;
  margin: 4px;
  padding: 4px;
}

.c1 {
  font-family: "Oleo Script",cursive;
  font-size: 24px;
  margin: 0;
  padding: 0;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0 c1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Adeola Ade
</a>
`;

exports[`Testing components Nav 1`] = `
.c1 {
  color: #008080;
  cursor: pointer;
  margin: 4px;
  padding: 4px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

@media screen and (max-width:768px) {
  .c0 {
    width: 100%;
  }
}

<nav
  className="c0"
>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
</nav>
`;

exports[`Testing components Renders links appropriately 1`] = `
.c0 {
  color: #008080;
  cursor: pointer;
  margin: 4px;
  padding: 4px;
}

<a
  className="c0"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  About
</a>
`;
